libvast/CMakeLists.txt

```diff
@@ -168,105 +172,114 @@ if (NOT VAST_ENABLE_BUNDLED_CAF)
+      libcaf_core_${_linkage_suffix}
+      PROPERTIES EXPORT_NAME core
+                 CXX_STANDARD 17
+                 CXX_STANDARD_REQUIRED ON
+                 CXX_EXTENSIONS OFF)
+    add_library(caf::core ALIAS libcaf_core_${_linkage_suffix})
+    target_compile_features(libcaf_core_${_linkage_suffix} INTERFACE cxx_std_17)
+    set_target_properties(libcaf_io_${_linkage_suffix} PROPERTIES EXPORT_NAME io)
+    target_compile_options(
+      libcaf_io_${_linkage_suffix} PRIVATE -Wno-maybe-uninitialized
+                                           -Wno-unknown-warning-option)
+    find_package(Threads REQUIRED)
+    target_link_libraries(libcaf_io_${_linkage_suffix} Threads::Threads)
+    add_library(caf::io ALIAS libcaf_io_${_linkage_suffix})
     install(
-      TARGETS libcaf_openssl_${_linkage_suffix}
+      TARGETS libcaf_core_${_linkage_suffix} libcaf_io_${_linkage_suffix}
       EXPORT CAFTargets
       ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
       LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
       RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
-    set(VAST_USE_OPENSSL true)
-  endif ()
-  add_library(libcaf_test INTERFACE)
-  target_link_libraries(libcaf_test INTERFACE caf::core)
-  target_include_directories(
-    libcaf_test
-    INTERFACE
-      "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/aux/caf/libcaf_test>"
-      "$<INSTALL_INTERFACE:${CMAKE_INSTALL_FULL_INCLUDEDIR}>")
-  set_target_properties(libcaf_test PROPERTIES EXPORT_NAME test)
-  add_library(caf::test ALIAS libcaf_test)
-  install(
-    TARGETS libcaf_test
-    EXPORT CAFTargets
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR})
-  set(caf_dir ${CMAKE_CURRENT_SOURCE_DIR}/aux/caf)
-  export(
-    EXPORT CAFTargets
-    FILE CAFTargets.cmake
-    NAMESPACE caf::)
-  install(
-    EXPORT CAFTargets
-    DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}/cmake/vast"
-    NAMESPACE caf::)
-  string(APPEND VAST_EXTRA_TARGETS_FILES
-         "\ninclude(\"\${CMAKE_CURRENT_LIST_DIR}/CAFTargets.cmake\")")
-  set(CAF_FOUND true)
-else ()
-  message(
-    FATAL_ERROR
-      "CAF not found, either use -DCAF_ROOT_DIR=... or initialize the libvast/aux/caf submodule"
-  )
+    if (VAST_ENABLE_OPENSSL)
+      target_include_directories(
+        libcaf_openssl_${_linkage_suffix}
+        PUBLIC
+          $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/aux/caf/libcaf_openssl>
+          $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+      set_target_properties(libcaf_openssl_${_linkage_suffix}
+                            PROPERTIES EXPORT_NAME openssl)
+      add_library(caf::openssl ALIAS libcaf_openssl_${_linkage_suffix})
+      install(
+        TARGETS libcaf_openssl_${_linkage_suffix}
+        EXPORT CAFTargets
+        ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+        RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
```

Comment:
Why do we use `CMAKE_INSTALL_BINDIR` here and below `*FULL_BINDIR`?

Reply:
An oversight from the previous PR that was missed.

FYI: CMake prepends `CMAKE_INSTALL_PREFIX` if the given path is relative. `CMAKE_INSTALL_FULL_*DIR` is calculated in the exact same way, so technically it doesn't make a difference.

Reply:
Should we at least use a one consistent version?